### ISSUE TITLE
Lights Turn Off

### ISSUE THREAD

**Issue by fotis3d** [rospogrigio/localtuya#1019]
After latest update I noticed after HA restart and also Local Tuya intergration reload lights connected with Local Tuya turn off.

Anyone else ?

**Comment by sibowler** [comment]
There is some new logic in the code to restore device state after HA restarts (where HA saves the current state every 15 mins or so). However the states wouldn't have been saved prior to the code coming in, so may have triggered a turn-off after the initial restart. 

Does this happen every restart / reload?

**Comment by fotis3d** [comment]
Yeah on every restart of HA or local tuya reload.

**Comment by sibowler** [comment]
That's puzzling - there's another piece of code in this release which initialises entities with a default/last value when Tuya doesn't provide a status value (Passive DPS Devices). But that should only be activated if the "Restore the last set value in HomeAssistant after a lost connection" setting is ticked under the entity settings. 

By default that setting _should_ be off. Could you double check the localtuya entity setup for the light to make sure that option is not ticked (it's at the bottom of the entity setting)

**Comment by LS2021-dev** [comment]
I‘ve got the same problem

**Comment by fotis3d** [comment]
I do not have such setting anywhere or I am so dumb I cannot find it.

![Screenshot_20220907-002611_Home Assistant](https://user-images.githubusercontent.com/42974039/188743066-2b07a254-85e8-4654-abd1-29c961a3a058.jpg)

**Comment by Fma965** [comment]
Same issue here after updating to the latest version

**Comment by LS2021-dev** [comment]
> I do not have such setting anywhere or I am so dumb I cannot find it.
> 
> 
> 
> ![Screenshot_20220907-002611_Home Assistant](https://user-images.githubusercontent.com/42974039/188743066-2b07a254-85e8-4654-abd1-29c961a3a058.jpg)
> 
> 

The setting is in integrations-localtuya-configure-edit device. You need to select the device and go to the bottom of the last page.

**Comment by Fma965** [comment]
It doesn't exist mate
![image](https://user-images.githubusercontent.com/1707290/188745791-e4858785-eced-4e62-a0a3-cde7639ea41c.png)

**Comment by fotis3d** [comment]
> The setting is in integrations-localtuya-configure-edit device. You need to select the device and go to the bottom of the last page.


If you mean these last also empty. 

![Screenshot_20220907-072007_Home Assistant](https://user-images.githubusercontent.com/42974039/188788237-40c41a17-fc1b-44b9-be91-fdf596b66d1c.jpg)

**Comment by fokcuk** [comment]
following this as my power sockets turn off - not good for a fridge =(

**Comment by fotis3d** [comment]
My sockets fortunately did not have this but because of the fear and I also have some in critical spots such as fridge I rolled back a version and everything is good.

**Comment by LS2021-dev** [comment]
> It doesn't exist mate ![image](https://user-images.githubusercontent.com/1707290/188745791-e4858785-eced-4e62-a0a3-cde7639ea41c.png)

I think the setting only exists for sockets
![image](https://user-images.githubusercontent.com/86912875/188835731-93d35d5e-60af-4721-951e-f30d1da0554e.png)

**Comment by Fma965** [comment]
> > It doesn't exist mate ![image](https://user-images.githubusercontent.com/1707290/188745791-e4858785-eced-4e62-a0a3-cde7639ea41c.png)
> 
> I think the setting only exists for sockets
> ![image](https://user-images.githubusercontent.com/86912875/188835731-93d35d5e-60af-4721-951e-f30d1da0554e.png)
> 

Then surely it defaults to off, either way this is a bug and I'm sure the dev will look in to it soon :)

**Comment by sibowler** [comment]
Yep - i've been thinking about it more, and I think it might be the new RESET command. Ironically this code has probably been tested the most (was sourced from 2 different PRs), but I believe it might be responsible for the behavior we're seeing here.

I'll write up a change to soft-disable it, so that it doesn't get used unless the RESET parameters are setup in the device setup and see if that makes a difference.

**Comment by sibowler** [comment]
I've made a change in PR #1022 - if you could test that out, that would be appreciated.

**Comment by sibowler** [comment]
> following this as my power sockets turn off - not good for a fridge =(

I think (or hope) what is occuring in your space is the RESET command is essentially confusing your device, meaning it doesnt report a value for DPS 1 (assuming this is the switch DPS). Then the new passive DPS code is recognising that you should have a status for DPS 1 and setting it to a default value (false), thus turning it off. 

Apart from the RESET fix (that i'm getting people to test), you can change the 'default fail-safe' to be **true** by editing the device through the localtuya "Edit Device" dialog flow. When you get to the switch config, where it says "Default value where un-initialised" enter _True_ here. This means that if HA ever can't receive a status (and doesn't have one saved), it will send a 'true' for the value.

**Comment by r17019** [comment]
I have noticed an issue which I think is likely related to this also.
My pet feeder has started to dispense food the same as if you selected '1' serving whenever HA or the localtuya integration is reloaded.
I believe it is likely due to the value being loaded at runtime. (It uses DP 3 for this and DP 15 to report back).
I have made the changes mentioned in PR #1022 but it doesn't seem to have made any difference. (After making the changes I restarted HA)
I haven't had a chance to remove and re-add the feeder yet though so I will try tomorrow and see if that makes a difference.
(Although my cat doesn't seem to mind)

**Comment by sibowler** [comment]
> I have noticed an issue which I think is likely related to this also. My pet feeder has started to dispense food the same as if you selected '1' serving whenever HA or the localtuya integration is reloaded. I believe it is likely due to the value being loaded at runtime. (It uses DP 3 for this and DP 15 to report back). I have made the changes mentioned in PR #1022 but it doesn't seem to have made any difference. (After making the changes I restarted HA) I haven't had a chance to remove and re-add the feeder yet though so I will try tomorrow and see if that makes a difference. (Although my cat doesn't seem to mind)

It might be related - When you re-test, if it's still occurring, can you raise this as a new issue?

**Comment by joe-sydney** [comment]
My floor fan turned on everytime I restarted HA....

**Comment by myevit** [comment]
same here. my garage door opener started to open on tuya device reconnect ( my wifi not stable in garage). rolling back. horrible logic.

**Comment by AndrewNovikov** [comment]
Same for me. Led lamp was switching on when wi-fi reconnect happened. This didn't happen with my second lamp, that's why I think this is a hardware issue. So I put in Aqara E1 switch to avoid accidental "light on" during the night. Now my lamp switching off after a 3 seconds delay when I switch it on. Started to see this behavior right after the update.

**Comment by sibowler** [comment]
So I've updated the PR #1022 to make the passive DPS restore logic explicit - i.e. an entity needs to be specified as a 'passive entity' for this to fire. This should fix all the issues with these other devices. Anyone who's interested in testing can find the updated code in the PR.

**Comment by r17019** [comment]
> So I've updated the PR #1022 to make the passive DPS restore logic explicit - i.e. an entity needs to be specified as a 'passive entity' for this to fire. This should fix all the issues with these other devices. Anyone who's interested in testing can find the updated code in the PR.

Looks good here, (my cat is less than impressed though).
Cheers

**Comment by jmcollin78** [comment]
> 

Hello I have such problems in my configuration. Can you please explains how to rollback the LocalTuya integration to 4.0.2 ?

**Comment by remlei** [comment]
> > 
> 
> Hello I have such problems in my configuration. Can you please explains how to rollback the LocalTuya integration to 4.0.2 ?

Same steps as you download it through HACS, by default the drop down selection is set to latest, just click that drop down selection abd that should list the oldsr versions, select a veraion you want to downgrade and click download.
